src/common/datasource/src/file_format/orc.rs

```diff
@@ -123,3 +139,93 @@ impl FileFormat for OrcFormat {
         Ok(schema)
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct OrcOpener {
+    object_store: Arc<ObjectStore>,
+    output_schema: SchemaRef,
+    projection: Option<Vec<usize>>,
+}
+
+impl OrcOpener {
+    /// Return a new [`OrcOpener`]. Fields indexes that are not in the `projection` will be ignored.
```

Comment:
` Fields indexes that are not in the projection will be ignored.` It seems like not right; it throws an error instead of being ignored 🥲